File: pl/c/temp.c
#include <stdio.h>

void fun(int a, float b) { printf("%d %.2f\n", a, b); }
void foo(int a, float b) { printf("XXXXX %d %.2f\n", a, b); }

int main() {

  void (*fp)(int, float);

  fp = fun;
  fp(1, 3.14);
  fp = foo;
  fp(1, 3.14);
}

File: pl/haskell/temp.hs
-- main :: IO ()
-- main = putStrLn (foo [1, 7, 8])

foo :: [Int] -> Int 
foo [] = 7
foo (x:xs) = if x `mod` 5 == 0 then foo xs else x + foo xs

main :: IO ()
main = print (foo [1, 7, 8])